### ISSUE TITLE
detached LUKS header is not waited for in generated systemd-cryptsetup@.service

### ISSUE THREAD

**Issue by rsteppleton** [systemd/systemd#10101]
**systemd version the issue has been seen with**
> 239

**Used distribution**
 > Arch

**Expected behaviour you didn't see**
 > If a `header` option is set in /etc/crypttab in the initramfs, a LUKS passphrase dialog should appear on boot via systemd-cryptsetup generated service

**Unexpected behaviour you saw**
 > The generated service fails with "Need a block device" error instead, causing dependency error and stopping boot.

**Steps to reproduce the problem**
Have a detached LUKS header on a block device that doesn't get initiated immediately at boot (such as USB stick - represented by [UUID2] below). Have line like this in `/etc/crypttab.initramfs`:
```
crypto /dev/disk/by-uuid/[UUID1] - header=/dev/disk/by-uuid/[UUID2]
```
Use `sd-encrypt` hook in mkinitramfs config and rebuild initramfs. Reboot. 

If the device gets initiated (i.e. its device nodes created) after the cryptsetup service is started, the service will fail. In my case, the device (a USB stick) gets initiated about one second after the service is started.

Apparently the service has a dependency on the [UUID1] device but not on [UUID2].

**Comment by poettering** [comment]
hmm, i figure this would be easy to add

**Comment by poettering** [comment]
Fixed in 38ab1ec9c8be5c4a728b3f27697eb9a3e4b37278.
